### PR TITLE
Kill event system

### PR DESCRIPTION
**Pull request by EternalWind**
The engine can be compiled successfully now. Some changes to the network
are still left to be done. For other parts, maybe there are also something
I havn't realized left but I think most of them are done though. In the
coming days, I will focus on modifying the tests and samples.

### CODE REVIEW

File: tests/src/TimerTest/TimerTest.hpp [hunk 1]
```diff
@@ -33,11 +33,14 @@ class Main : public dt::State {
...
 public:
     void OnInitialize();
-    void HandleEvent(std::shared_ptr<dt::Event> e);
+    //void HandleEvent(std::shared_ptr<dt::Event> e);
 
 public slots:
     void TimerCallback(const QString& message);
 
+private slots:
+    void _HandleEvent(double simulation_frame_time);
```
Comment (svenstaro) [inline tests/src/TimerTest/TimerTest.hpp:42]: I'd really like to get rid of HandleEvent completely. Why did you keep it? The timer should obviously lose its function to emit classic events and emit Qt signals instead.

**Comment by svenstaro** [comment]
What are you planning to use _HandleEvent for?

**Comment by EternalWind** [comment]
_HandleEvent is just a name, actually this method is a slot connected with the BeginFrame signal. You can name it _TestLogic, _HandleSignal or so though. I just thought the name _HandleEvent is not bad for describing what this method does.

**Comment by svenstaro** [comment]
Mh very well. I will merge for now but in general I'd prefer a better name. It sounds too generic right now. Something like OnTick or OnBeginFrame should be fine although we might also want to revise our whole OnX stuff.

**Pull request merged**